src/main/java/io/zipcoder/casino/GoFish.java

```diff
@@ -0,0 +1,202 @@
+package io.zipcoder.casino;
+
+import io.zipcoder.casino.utilities.Console;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class GoFish extends CardGame {
+
+    private Console c;
+    private GoFishDisplay d;
+    private Integer pairCountPlayer;
+    private Integer pairCountDealer;
+    private String rankRequested;
+    private String turn;
+
+    public GoFish() {
+
+        this.c = new Console(System.in, System.out);
+        this.d = new GoFishDisplay();
+
+        this.pairCountPlayer = 0;
+        this.pairCountDealer = 0;
+        this.turn = "player";
+        this.rankRequested = "";
+    }
+
+    public String getTurn() {
+        return this.turn;
+    }
+
+    public void setTurn(String newTurn) {
+        this.turn = newTurn;
+    }
+
+    public String getRankRequested() {
+        return this.rankRequested;
+    }
+
+    public void setBookCounts(int bookCountPlayer, int bookCountDealer) {
+        this.pairCountPlayer = bookCountPlayer;
+        this.pairCountDealer = bookCountDealer;
+    }
+
+    public int getPairCountPlayer() {
+        return this.pairCountPlayer;
+    }
+
+    public int getPairCountDealer() {
+        return this.pairCountDealer;
+    }
+
+    public void setRankRequested(String rank) {
+        this.rankRequested = rank;
+    }
+
+    public void setupGame(int numberOfCards) {
+        this.createNewDeck();
+        this.shuffleCards();
+        this.dealCards(numberOfCards);
+    }
+
+
+    /*public void getRankToAskFor_Human() {
+    // Move to display class??
+    public String displayPlayersHand(ArrayList<String> playersHand) {
+        String playersHandOutput = "";
+        Iterator itr = playersHand.iterator();
+        while(itr.hasNext()) {
+            playersHandOutput += itr.next() + "\n";
+        }
+        return playersHandOutput;
+    }*/
+
+    public void getRankToAskFor_Human() {
+
+        d.printWhatRankYouWillAskFor();
+        this.rankRequested = c.getStringInput("", null);
+    }
+
+    public void getRankToAskFor_Computer() {
+       int pickCard = (int) Math.random() * (this.dealersHand.size() - 1);
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [RV: Random value from 0 to 1 is coerced to the integer 0 (RV_01_TO_INT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#rv-random-value-from-0-to-1-is-coerced-to-the-integer-0-rv-01-to-int).
A random value from 0 to 1 is being coerced to the integer value 0. You probably want to multiply the random value by something else before coercing it to an integer, or use the Random.nextInt(n) method.